### PR TITLE
add test makethin

### PR DESCRIPTION
**Pull request by rdemaria**
trying to test makethin...

### CODE REVIEW

**Comment by coldfix** [comment]
Good thinking to add a regression test.

I would suggest to replace the python class with a method of `TestMadx` as follows:

```python
class TestMadx:

    ...

    def test_makethin(self):
        # regression test for segfault, see #67:
        self.mad.chdir(os.path.dirname(__file__))
        self.mad.call('test_makethin.madx')
```

File: test/test_madx.py [hunk 1]
```diff
@@ -851,6 +851,11 @@ def test_subrange(self):
             beam;
         """)
 
+    def test_makethin(self):
```
Comment (coldfix) [inline test/test_madx.py:854]: Make this a method of `TestMadx` (it's currently under `TestTransfermap`)

**Review by coldfix — APPROVED** [review]

**Comment by coldfix** [comment]
Thanks, remaining issue was due to permission problem (uploading coverage results from within pull request currently not possible).

**Comment by rdemaria** [comment]
Thank to you!

I saw that madx.sh still contains -flto. Do I understand correctly that the lto pass does not apply when building the libmadx.a ?

**Comment by coldfix** [comment]
As far as I understand: passing `-flto` at the compile stage instructs the compiler to generate `.o/.a` output files that are compatible with link time optimization (LTO). At the link stage, if any of the linked object files was compiled with `-flto`, the linker will by default use LTO, which can be disabled by passing `-fno-flto`.

So there are two ways to disable LTO:

- build madx's `libmadx.a` without `-flto` in the first place
- link cpymad's `libmadx.so` with `-fno-lto`

It doesn't really matter that much which option we choose, because passing `-flto` doesn't really increase the compile time itself (only link time if LTO is enabled). On the other hand, passing `-flto` at compile time makes it possible to link the same `libmadx.a` with LTO enabled or disabled, which means that it's easier to try cpymad in both modes without recompiling MAD-X. That's why I suggested to go the second route rather than removing `-flto` in both files.

**Pull request merged**